_posts/2018-08-03-metrics.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+title: Refining the Isolation Metric
+---
+
+Nabla containers are designed around the idea that thinner, 
+lower-level interfaces between the guest containers and the (shared 
+and privileged) host kernel result in better isolation for tenants in
+the cloud.  The reasoning is, when comparing to fatter, more
+high-level interfaces: (1) thinner interfaces are assumed to be more
+easily guarded, and (2) the need to only implement lower-level
+abstractions leads to less complexity in the underlying layer, which,
+in turn, leads to a lower probability of bugs that can ultimately
+result in the compromise of the isolation implemented by the
+underlying layer.  These arguments are not new: we are all familiar
+with similar arguments involving hypervisors and VM isolation.[^1]
+
+Unfortunately, terms like "thin interface" or "low-level interface"
+are somewhat subjective.  We have been working on various ways to
+quantify it in different scenarios, which we refer to as our
+"isolation metric".
+
+### Counting kernel code
+
+Our first stab at an isolation metric is based on counting the amount
+kernel code needed to support various containers.  We make the
+assumption that the less kernel code we observe is a proxy for less
+complexity, which is a proxy for less bugs/vulnerabilities, which is a
+proxy for greater assurance that the underlying kernel code is
+maintaining isolation.
+
+We use the kernel's `ftrace` mechanism to trace container-related
```

Comment:
Can we add a link to ftrace https://lwn.net/Articles/365835/ , or https://elinux.org/Ftrace